Cython/Shadow.pyi

```diff
@@ -116,11 +118,11 @@ class _EmptyDecoratorAndManager(object):
         tb: Optional[TracebackType]
     ) -> None: ...
 
-cclass = ccall = cfunc = _EmptyDecoratorAndManager()
+cclass = ccall = cfunc = inline = _EmptyDecoratorAndManager()
 
 def declare(
     t: Optional[Callable[..., _T]] = None,
     value: Any = ...,
     *,
     visibility: str = ...
-) -> Optional[_T]: ...
+) -> _T: ...
```

Comment:
huh? it literally  can return `None`

[look](https://github.com/cython/cython/blob/8af0271186cc642436306274564986888d5e64c8/Cython/Shadow.py#L199)

Reply:
That happens if `_T` is not a type, or `_T` is `None`.

The former I is guarded by the typehint, and if they ignore that there's nothing we can do. But for the latter, `_T` will be `None` so the return type becomes `_T`.

Or am I misunderstanding something here?

Reply:
Yes, you are right, I did not think about it that way.

But there is a possibility that `cast` will return the first argument and not the result of the function call. [Look](https://github.com/cython/cython/blob/8af0271186cc642436306274564986888d5e64c8/Cython/Shadow.py#L175)

We probably need to take this into account ..

Reply:
I don't remember how the code looked like when this comment was made, but I think I solved it. Looking at the comment and code again I did actually realize that there was a(nother?) problem which I fixed in 6c976ec3c1d0c78679e8a4153eb2e0d02bb3c914.